### ISSUE TITLE
RuntimeException / Unknown column 't1.o_id' in 'where clause'

### ISSUE THREAD

**Issue by mwjames** [SemanticMediaWiki/SemanticMediaWiki#1222]
This just appeared during a different test for when property `HasWorkManifested` was defined as `[[Has type::Text]]` and executed on:

```
[[HasWorkManifested::ABC]]
{{#show:{{FULLPAGENAME}}|?-HasWorkManifested|link=none}}
{{#show:{{FULLPAGENAME}}|?HasWorkManifested|link=none}}
```

`smw_di_blob` doesn't have a `o_id` column so its obvious that the query must fail.

```
Query: SELECT DISTINCT smw_title, smw_namespace, smw_iw, smw_sortkey, smw_subobject FROM `abcsmw_object_ids` INNER JOIN `abccsmw_di_blob` AS t1 ON t1.s_id=smw_id WHERE t1.p_id='10201' AND t1.o_id='10196' 
Function: SMWSQLStore3Readers::getPropertySubjects
Error: 1054 Unknown column 't1.o_id' in 'where clause' (localhost)

#0 ...\includes\db\Database.php(1205): DatabaseBase->reportQueryError('Unknown column ...', 1054, 'SELECT DISTINC...', 'SMWSQLStore3Rea...', false)
#1 ...\includes\db\Database.php(1726): DatabaseBase->query('SELECT DISTINC...', 'SMWSQLStore3Rea...')
#2 ...\extensions\SemanticMediaWiki\src\MediaWiki\Database.php(179): DatabaseBase->select('`abcsmw_object_...', 'DISTINCT smw_ti...', 't1.p_id='10201'...', 'SMWSQLStore3Rea...', Array, Array)
#3 ...\extensions\SemanticMediaWiki\includes\storage\SQLStore\SMW_SQLStore3_Readers.php(461): SMW\MediaWiki\Database->select('`abcsmw_object_...', 'DISTINCT smw_ti...', 't1.p_id='10201'...', 'SMWSQLStore3Rea...', Array)
#4 ...\extensions\SemanticMediaWiki\includes\storage\SQLStore\SMW_SQLStore3_Readers.php(197): SMWSQLStore3Readers->getPropertySubjects(Object(SMW\DIProperty), Object(SMW\DIWikiPage), NULL)
#5 ...\extensions\SemanticMediaWiki\src\SQLStore\Lookup\CachedValueLookupStore.php(257): SMWSQLStore3Readers->getPropertyValues(Object(SMW\DIWikiPage), Object(SMW\DIProperty), NULL)
#6 ...\extensions\SemanticMediaWiki\includes\storage\SQLStore\SMW_SQLStore3.php(262): SMW\SQLStore\Lookup\CachedValueLookupStore->getPropertyValues(Object(SMW\DIWikiPage), Object(SMW\DIProperty), NULL)
#7 ...\extensions\SemanticMediaWiki\includes\storage\SMW_ResultArray.php(219): SMWSQLStore3->getPropertyValues(Object(SMW\DIWikiPage), Object(SMW\DIProperty), NULL)
#8 ...\extensions\SemanticMediaWiki\includes\storage\SMW_ResultArray.php(110): SMWResultArray->loadContent()
#9 ...\extensions\SemanticMediaWiki\includes\storage\SMW_ResultArray.php(139): SMWResultArray->getNextDataItem()
#10 ...\extensions\SemanticMediaWiki\includes\queryprinters\ListResultPrinter.php(343): SMWResultArray->getNextDataValue()
#11 ...\extensions\SemanticMediaWiki\includes\queryprinters\ListResultPrinter.php(286): SMW\ListResultPrinter->getRowListContent(Array)
#12 ...\extensions\SemanticMediaWiki\includes\queryprinters\ListResultPrinter.php(170): SMW\ListResultPrinter->getRowText(Array, Object(SMWQueryResult))
#13 ...\extensions\SemanticMediaWiki\includes\queryprinters\ResultPrinter.php(250): SMW\ListResultPrinter->getResultText(Object(SMWQueryResult), 2)
#14 ...\extensions\SemanticMediaWiki\includes\queryprinters\ResultPrinter.php(215): SMW\ResultPrinter->buildResult(Object(SMWQueryResult))
#15 ...\extensions\SemanticMediaWiki\includes\query\SMW_QueryProcessor.php(516): SMW\ResultPrinter->getResult(Object(SMWQueryResult), Array, 2)
#16 ...\extensions\SemanticMediaWiki\includes\parserhooks\AskParserFunction.php(176): 
```

This also closes #400 which reported the same issue but without clear instructions on how to replicate the issue.

**Comment by mwjames** [comment]
This is not a regression as the code in `SMWSQLStore3Readers::prepareValueQuery` has been updated in any recent release (at least not since 1.9).

**Comment by mwjames** [comment]
Fixed with ca32ff4.
